pkg/storage/database/database.go

```diff
@@ -18,19 +15,13 @@ type Database interface {
 	GetDestinationCredentials(ctx context.Context, dbID int64) (config.Destination, error)
 
 	AddAPIKey(ctx context.Context, destId int64, hashedAPIKey string) error
-	GetAPIKeyDetails(ctx context.Context, hashedAPIKey string) (models.APIKey, error)
+	GetAPIKeyDetails(ctx context.Context, hashedAPIKey string) (APIKey, error)
 
 	CreateShareQuery(ctx context.Context, destId int64, query string, expires time.Duration) (queryId uuid.UUID, err error)
-	GetShareQuery(ctx context.Context, queryId uuid.UUID) (models.SharedQuery, bool)
+	GetShareQuery(ctx context.Context, queryId uuid.UUID) (SharedQuery, bool)
 
-	Hash(s string) string
-}
-
-func NewDatabaseConnection(conf config.Database, destinations []config.Destination, adminKeys []config.APIKey) Database {
```

Comment:
This is the behavior I want:

- If the user is configuring Scratch via config file, then we should only read values from that config. (Destinations, API Keys, etc.) We could even name this database "config" or "static". In this case, the user _cannot_ add a new destination. This is so there is a very clear single source of truth for where data is coming from - either a database or a config file. 
- Otherwise, read _all_ values from the database, and ignore any destinations or API keys defined in the config.
- Let's decouple Gorm. We can have a common Gorm implementation or set of utilities but inevitably sqlite and postgres will have differences.
- In the future, I might want to add dynamodb or firestore so that this can be deployed as a completely serverless solution. So decoupling from Gorm is the way to go.

That means